app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt

```diff
@@ -739,10 +722,38 @@ class DefaultSessionControlControllerTest {
             every { id } returns R.id.homeFragment
         }
 
-        controller.handleMenuOpened()
+        createController().handleMenuOpened()
 
         verify(exactly = 0) {
             navController.navigateUp()
         }
     }
+
+    private fun createController(
```

Comment:
I am curious about the reasoning  and lessons that can be learned behind adding a `createController` and calling it for each test. Does that help with the intermittent? Just looking at the issue comment, I assume we were just removing the mock for lambdas to help avoid intermittents.

Reply:
@gabrielluong yeah, that's all there is to it. There's a problem with `mockk` mocking lambdas that we're working around here for now. This started happening with Java 11 and we didn't want to back that out and all related upgrades that were needed for jetpack compose. We also can't wait until an upstream fix is available because there were too many intermittents to make this feasible. So, we can only work around it for now. We have it on the agenda for our next Android catch up too to decide what to do and how to prevent us from introducing more of those.